### ISSUE TITLE
Activities missing in Activity History/ PGCR

### ISSUE THREAD

**Issue by BlazeBoss17** [Bungie-net/api#867]
We've been seeing a lot of reports from our users that completed activities are not showing up in either activity history or as postgames. I normally wouldn't think much of these reports, but I believe there is an actual issue right now due to the volume of these reports across many different users and platforms. 

I have seen a few youtube videos of runs that have occurred in the last couple days that are nowhere to be found in either activity history or a PGCR.

**Comment by vthornheart-bng** [comment]
Hmm, could you send me one of these users' membership IDs, and ideally if you can one of these videos?

**Comment by BlazeBoss17** [comment]
4611686018429538058 claims to be missing a flawless scourge of the past on his warlock on Tuesday that unlocked the diamond triumph but isn't showing up on the API.
*I lost the link to this video but can add it later

4611686018484557780 claims to be missing a 2 man eater of worlds sometime earlier this week.
https://www.youtube.com/watch?v=muCJGjfN1Zs&feature=youtu.be

**Comment by BlazeBoss17** [comment]
Twitch stream of the first raid from the perspective of player 4611686018435401909: https://www.twitch.tv/videos/371845823?t=06h38m13s

**Comment by vthornheart-bng** [comment]
Were any of these the one that *8058 was looking for?

https://www.bungie.net/en/PGCR/3553038255?character=2305843009260578095

  https://www.bungie.net/en/PGCR/3552881569?character=2305843009260578095

  https://www.bungie.net/en/PGCR/3548504070?character=2305843009277918128

**Comment by BlazeBoss17** [comment]
No, unfortunately not. From what I can put together, here are the activities 4611686018429538058 played on Jan. 29:

| Activity | InstanceId | Character | Period (Mountain) |
|----------|------------|-----------|-------------------|
|Scorge of the Past|	3552430180|	T|	2:57|
|Scorge of the Past|	3552633695|	H|	3:34|
|Scorge of the Past|	3552881569|	W|	4:03|
|Scorge of the Past|	3553038255|	W|	4:42|
|Scorge of the Past|	???|	W|	???|
|EDZ Patrol|	3553238185|	H|	5:30|
|Shattered Throne|	3553405876|	H|	5:35|
|Shattered Throne|	3553573959|	T|	6:03|
|Shattered Throne|	3553877541|	W|	6:50|

**Comment by vthornheart-bng** [comment]
Ah, good info - thank you!  I'll hit up some folks.

**Comment by floatingatoll** [comment]
"Completed activities" — can you expand on this phrase — do you normally
only look at raids, or do you also look at patrol? Are users reporting only
missing raids, or also strikes/crucible/? etc.

**Comment by BlazeBoss17** [comment]
Our website, raid.report, only shows raids, so that is exclusively what user feedback has complained about.

**Comment by BlazeBoss17** [comment]
We are still receiving consistent complaints about activities never showing up. This seems like a large issue as data is potentially being lost forever.

**Comment by vthornheart-bng** [comment]
I'll hit them up again, I've got no further info on the issue as of this time.  Did this start happening as of Feb 1st, or were you seeing it before that?

Users who track other activities, are you seeing this for anything other than Raids?

**Comment by BlazeBoss17** [comment]
We started seeing a consistent volume of claimplaints after January 29. Not saying it wasn’t happening before that, but that was when I noticed it.

**Comment by vthornheart-bng** [comment]
Cool, good deal.  All this info is useful, thank you!  Hopefully we can get you some more data soon - I hit up the stats folks and we'll see what their investigation brings up when they're able to do so.

**Comment by CrunchyWizard47** [comment]
I've got a troublesome activity where I've got three clan members participating in an Iron Banner Control on 3/28 (2019-03-29T02:41:48Z) with an instanceId of 3805062447. It normally doesn't show in the activity list of any of the clan members, but it's in guardian.gg (that's how I found it) and I've had it pop up once or twice (out of about 40 attempts) when hitting https://stats.bungie.net/Platform/Destiny2/1/Account/4611686018443078628/Character/2305843009267861329/Stats/Activities/?count=200&mode=0&page=0 but mostly it doesn't appear...
We require that all clan members play in at least one fireteam with other clan members every calendar month or they get kicked. This member was kicked because the activity didn't show. He protested and that's how I found this. This activity normally doesn't show when I run the activity list of any of the three members: (membershipId/characterId: 4611686018443078628/2305843009267861329, 4611686018461296896/2305843009266719951, 4611686018463061258/2305843009268766526). 

Hope this helps track it down...

**Comment by vthornheart-bng** [comment]
In your case @CrunchyWizard47, were you checking more than the first page of results?  It could be that some of those users have it on a later page.

**Comment by CrunchyWizard47** [comment]
@vthornheart-bng - yes, my script iterates through all activities that are returned for a player and then asks for the PGCR for those which it doesn't already have in my database. My experience using the URL in a browser is that I've seen the instanceId show up only once over around 30 tries (while I don't depend on the activities showing up in chronological order, I find they do and for the URL above, the activities go back to 2018). I understand it's probably caching most of those (and I don't have access to any of those accounts to 'break' the cache) but I've run my script many times (maybe 20?) over the past week and it hasn't capturedthat instanceId yet. And this is an activity in which three of my members took part, so it should have popped up three times each time I've run it.

**Comment by vthornheart-bng** [comment]
Hmm, that is an interesting issue.  Indeed, they are meant to come down in chronological order.

I'm a bit pressed for time at the moment, but I'll see if I can dig into it a little more when I find some time to go digging into the databases.

**Comment by vthornheart-bng** [comment]
Mind if we roll this into a different Github issue?  I think this is presenting a different bug than the one mentioned above, it'll be good to track them separately.

**Comment by CrunchyWizard47** [comment]
That's fine by me. I appreciate your attention.

**Comment by vthornheart-bng** [comment]
No prob, I appreciate the info!  Hopefully we can get to the bottom of this.

Moved it over to:

https://github.com/Bungie-net/api/issues/904
